custom_components/tahoma/water_heater_devices/domestic_hot_water_production.py

```diff
@@ -51,13 +74,20 @@ def max_temp(self):
     def current_operation(self):
         """Return current operation ie. eco, electric, performance, ..."""
         return OVERKIZ_TO_OPERATION_MODE[
-            self.executor.select_state(OverkizState.IO_DHW_MODE)
+            OVERKIZ_TO_OPERATION_MODE[OverkizCommandParam.BOOST]
+            if self._is_boost_mode_on
+            else self.executor.select_state(
+                OverkizState.IO_DHW_MODE, OverkizState.MODBUSLINK_DHW_MODE
+            )
```

Comment:
Can this be written a bit different (on multiple lines?) so it is more clear. I don't understand how te result of `OVERKIZ_TO_OPERATION_MODE[]` can be passed to `OVERKIZ_TO_OPERATION_MODE[]` by the way. Does this just work since we are currently using the same values on left and right side?

Reply:
fixed a typo, is it better now?